File: tfginfo/src/webapi/Modules/Student/StudentModel.cs
using System.Text.Json.Serialization;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;

namespace TFGinfo.Models
{
    public class StudentModel
    {
        public int id { get; set; }
        public string name { get; set; }
        public string surname { get; set; }
        public string email { get; set; }
        public string dni { get; set; }
        public string? phone { get; set; }
        public string? address { get; set; }
        public int career { get; set; }
        public int user { get; set; }
        public DateTime? birthdate { get; set; }

        [JsonIgnore]
        public CareerModel careerModel { get; set; }

        public List<TFGStudentModel> TFGs { get; set; }
    }

    public class StudentConfiguration : IEntityTypeConfiguration<StudentModel>
    {
        public void Configure(EntityTypeBuilder<StudentModel> builder)
        {
            builder.HasOne(p => p.careerModel)
                   .WithMany(u => u.Students)
                   .HasForeignKey(p => p.career)
                   .OnDelete(DeleteBehavior.Cascade);
        }
    }

}

File: tfginfo/src/webapp/src/app/modules/admin/models/student.model.ts
export interface StudentBase {
    id?: number;
    name: string;
    surname: string;
    email: string;
    dni: string;
    phone?: string;
    address?: string;
    birthdate?: Date;
}

export interface StudentDTO extends StudentBase {
    career: CareerBase;
}

export interface StudentFlatDTO extends StudentBase {
    careerId: number;
}

export interface CareerBase {
    id: number;
    name: string;
}